lib/packwerk/package_set.rb

```diff
@@ -38,29 +45,40 @@ def package_paths(root_path, package_pathspec)
 
       private
 
+      sig { params(packages: T::Array[Package]).void }
       def create_root_package_if_none_in(packages)
         return if packages.any?(&:root?)
         packages << Package.new(name: Package::ROOT_PACKAGE_NAME, config: nil)
       end
     end
 
+    sig { params(packages: T::Array[Package]).void }
     def initialize(packages)
       # We want to match more specific paths first
       sorted_packages = packages.sort_by { |package| -package.name.length }
       @packages = sorted_packages.each_with_object({}) { |package, hash| hash[package.name] = package }
     end
 
+    sig { override.params(blk: T.proc.params(arg0: Package).returns(T.untyped)).returns(T.untyped)}
     def each(&blk)
-      @packages.values.each(&blk)
+      packages.values.each(&blk)
     end
 
+    sig { params(name: String).returns(T.nilable(Package)) }
     def fetch(name)
-      @packages[name]
+      packages[name]
     end
 
+    sig { params(file_path: T.any(Pathname, String)).returns(T.nilable(Package)) }
     def package_from_path(file_path)
       path_string = file_path.to_s
-      @packages.values.find { |package| package.package_path?(path_string) }
+      packages.values.find { |package| package.package_path?(path_string) }
+    end
+
+    sig { returns(T::Hash[String, Package]) }
+    def packages
```

Comment:
I pull this into a separate method so I can properly initialize the instance variable as nilable, while clients of this method will always be granted a non-nilable type.

Reply:
sure, changed!